packages/headless/src/controllers/insight/did-you-mean/headless-insight-did-you-mean.ts

```diff
@@ -10,20 +10,33 @@ import {
   buildCoreDidYouMean,
   DidYouMean,
   DidYouMeanState,
+  DidYouMeanProps,
+  DidYouMeanOptions,
 } from '../../core/did-you-mean/headless-core-did-you-mean.js';
 
-export type {QueryCorrection, WordCorrection, DidYouMean, DidYouMeanState};
+export type {
+  QueryCorrection,
+  WordCorrection,
+  DidYouMean,
+  DidYouMeanState,
+  DidYouMeanProps,
+  DidYouMeanOptions,
+};
 
 /**
  * The insight DidYouMean controller is responsible for handling query corrections.
  * When a query returns no result but finds a possible query correction, the controller either suggests the correction or
  * automatically triggers a new query with the suggested term.
  *
  * @param engine - The insight engine.
+ * @param props - The configurable `DidYouMean` properties.
  */
-export function buildDidYouMean(engine: InsightEngine): DidYouMean {
+export function buildDidYouMean(
+  engine: InsightEngine,
+  props: DidYouMeanProps = {}
+): DidYouMean {
   const controller = buildCoreDidYouMean(engine, {
-    options: {queryCorrectionMode: 'legacy'},
+    options: {...props, queryCorrectionMode: 'legacy'},
```

Comment:
previously `queryCorrectionMode: 'legacy'` passed to force the legacy way to be used, now the did you mean options are passed so no need to force `legacy` otherwise the options passed will have no effect.

It would a good idea to cover this by a unit test that makes sure that `buildCoreDidYouMean` was called with the **correct** parameters.